imap/xapian_wrap.cpp

```diff
@@ -1621,17 +1626,17 @@ static Xapian::Query *query_new_listid(const xapian_db_t *db,
 }
 
 static Xapian::Query *query_new_messageid(const xapian_db_t *db __attribute__((unused)),
-                                          const char *prefix,
+                                          enum search_part partnum,
                                           const char *str)
 {
-    return new Xapian::Query(std::string(prefix) + str);
+    return new Xapian::Query(std::string(get_term_prefix(partnum)) + str);
```

Comment:
`get_term_prefix()` now returns a `std::string` so I think there's no need to construct another one here?

```suggestion
    return new Xapian::Query(get_term_prefix(partnum) + str);
```

Reply:
Right. Fixed.